### PR TITLE
Change directory structure to match the other repos

### PR DESCRIPTION
**Pull request by lionelvillard**
## Proposed Changes

- map source/* -> */source
- map sink/* -> */sink

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: config/sink/placeholder.go [hunk 1]
```diff
@@ -16,4 +16,4 @@ limitations under the License.
 
 // Package config is a placeholder that allows us to pull in config files
 // via go mod vendor.
-package config
+package sink
```
Comment (aavarghese) [inline config/sink/placeholder.go:19]: Why don't we keep the `package config` like we did for `source` [here](https://github.com/knative-sandbox/eventing-redis/blob/06f8afa58df477399adeff03d268f738126635eb/config/source/placeholder.go)?

Comment (aavarghese) [reply config/sink/placeholder.go:19]: @lionelvillard

Comment (lionelvillard) [reply config/sink/placeholder.go:19]: I ended up doing the opposite to be consistent with the other repos

**Review by aavarghese — APPROVED** [review]
Changes look good. 

Just leaving a note here that we'll need to make dir changes in https://github.com/knative-sandbox/eventing-autoscaler-keda/search?q=eventing-redis imports too.

/lgtm

**Pull request merged**